lib/server.js

```diff
@@ -196,7 +205,7 @@ module.exports = function createServer(config, log) {
       fetchRecords(ip, email, phoneNumber)
         .spread(
           function (ipRecord, reputation, emailRecord, ipEmailRecord, smsRecord) {
-            if (ipRecord.isBlocked()) {
+            if (ipRecord.isBlocked() && ! isWhitelisted(ip, email)) {
```

Comment:
The extra check is needed because we don't call `allowWhitelisted` in the following promise handler any more.